pkg/services/correlations/models.go

```diff
@@ -154,6 +154,9 @@ type UpdateCorrelationCommand struct {
 	// Optional description of the correlation
 	// example: Logs to Traces
 	Description *string `json:"description"`
+	// Correlation Configuration
+	// example: { field: "job", type: "query", target: { query: "job=app" } }
+	Config *CorrelationConfig `json:"config"`
```

Comment:
reusing `CorrelationConfig` here, where all members are marked as required, means we won't be able to update single fields in the config object

for instance, the following would fail with a 400:
```curl
curl --request PATCH \
  --url http://localhost:3000/api/datasources/uid/P8045C56BDA891CB2/correlations/wExVOJ44k \
  --header 'Authorization: Basic YWRtaW46YWRtaW4=' \
  --header 'Content-Type: application/json' \
  --data '{
	"config": {
		"target": {}
	}
}'
```

while updating only `config.target` should be allowed.

one possible approach could be to have a separate model for the update, something like
```go
// swagger:model
type CorrelationConfigUpdateDTO struct {
	// Field used to attach the correlation link
	Field *string `json:"field"`
	// Target type
	Type *CorrelationConfigType `json:"type"`
	// Target data query
	Target *map[string]interface{} `json:"target"`
}
```

and in the update handler merge non nil properties from this object into the ones stored in the DB.

Reply:
It's getting more complex to handle having API that allows individual properties to be updated this way. Is it how all Grafana APIs work? I wonder if it'd be just simpler to always pass the entire object with all properties to be updated).

Reply:
@Elfo404 As this wasn't mentioned in the issue I didn't implement it like you just suggested. I will be happy to implement it if we decide that we want to have this kind of complexity here (see question from @ifrost).

Reply:
That's how PATCH requests are meant to behave though, also this endpoint already works that way (you can update label or description independently) so i assumed not changing the behaviour was implicit.

we also have a few other PATCH endpoints that behave the same way by partially updating a resource ( ie. `/annotations/{uid}`)

if we want to allow updating the whole correlation i think it should be done in a PUT handler instead

Reply:
Right. BTW. are we planning to use partial updates on Correlations Page or we could actually have a `PUT` endpoint?

Reply:
technically we are already doing it as we never update the target datasource after creation (the PATCH endpoint also still does not support updating that property).

If we want to add a `PUT` endpoint then i think `targetUid` and `uid` should be handled as well given the semantics of a PUT request